applications/nrf_desktop/src/hw_interface/wheel.c

```diff
@@ -114,9 +114,16 @@ static void data_ready_handler(const struct device *dev, const struct sensor_tri
 		k_spin_unlock(&lock, key);
 	}
 
+	int err = sensor_sample_fetch_chan(qdec_dev, SENSOR_CHAN_ROTATION);
```

Comment:
Interesting to note is that this will have a side effect - the accumulate function in the driver will be called twice.
We might think of dropping the accumulate from handler in the driver: https://github.com/nrfconnect/sdk-zephyr/pull/2101/files#diff-a22294740c632db6ff7ba3f860c14a931d296505bebd45f68b23891c9d21e3bcL139

I am not sure this will be correct however as driver's handler works with data that is passed as an argument ot it. Likely we cannot change anything.

Reply:
From what I believe it is completely fine for accumulate function to be called multiple times since its primary role is to transfer values from smaller HW register to bigger 32-bit integer register